File: generators/app/templates/java/app/src/main/java/org/onepointzero/androidstarterjava/application/MainActivity.java
package <%= packageName %>.application;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import <%= packageName %>.R;
import <%= packageName %>.support.Bootstrap;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity implements MainScreen {

    @Inject MainPresenter presenter;

    @BindView(R.id.label_main)
    TextView labelMain;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ((Bootstrap) getApplication()).getComponent().inject(this);

        ButterKnife.bind(this);

        presenter.bind(this);
    }

    @Override
    public void showLabel(String message) {
        labelMain.setText(message);
    }
}

File: generators/app/templates/kotlin/app/src/main/java/org/onepointzero/androidstarterkotlin/application/MainActivity.kt
package <%= packageName %>.application

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import <%= packageName %>.R
import <%= packageName %>.support.app
import javax.inject.Inject

class MainActivity : AppCompatActivity(), MainScreen {

    private val component by lazy { app.component }

    @Inject lateinit var presenter: MainPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        component.inject(this)
        presenter.bind(this)
    }

    override fun printText(text: String) {
        mainLabel.text = text
    }
}
